### PR TITLE
Polyfill useId rather than random id

### PR DESCRIPTION
**Pull request by MikeKingdom**
UPDATE: Tested with React 17.

#### What does this PR do?
The change we made to avoid React 18's `useId` #7351 has the downside that it makes test snapshots involving Accordions change randomly due to the randomly generated button ids. It also has a downside in that the id for an Accordion button changes every render.

This change instead creates a polyfill for React 18's `useId` which works much like their `useId` and stays consistent for tests and renders. If React 18 is being used, it will simply use the React `useId` implementation instead.
  
#### Where should the reviewer start?
utils/useId.js

#### What testing has been done on this PR?
Basic testing with storybook and simulating React 18 being available. It could use a little more testing with the actual React 17 or 16.

#### How should this be manually tested?
Storybook accordion stories.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible although it can change existing snapshots.

### CODE REVIEW

**Comment by britt6612** [comment]
one small comment everything else LGTM

File: src/js/utils/useId.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useId as useIdReact, useState } from 'react';
+
+let currentId = 0;
+
+const getId = () => {
+  // eslint-disable-next-line no-plusplus
+  const id = currentId++;
```
[the remark this answers is not in the document]
Comment (MikeKingdom) [reply src/js/utils/useId.js:7]: Can if you prefer. I'm not a fan of that eslint rule and if you read what that rule is trying to protect against, this isn't really the problem situation, especially when we require ;

[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/utils/useId.js:7]: it was more of a soft suggestion I wouldn't mind if we left it just knew we could do it that way as well

**Review by taysea — APPROVED** [review]
Looks good to me! I know you mentioned considering adding `r` to the id to align with React18/minimize snapshot changes down the line. If you want to do that I'm supportive, otherwise I'm fine with how it is.

**Review by britt6612 — APPROVED** [review]

**Review by halocline — APPROVED** [review]

**Comment by jcfilben** [comment]
With this change we should also be able to remove the `beforeEach` and `afterEach` blocks from Accordion-test

**Comment by MikeKingdom** [comment]
> With this change we should also be able to remove the `beforeEach` and `afterEach` blocks from Accordion-test

Good point!

**Comment by muddarla-vijaya-durga** [comment]
may i know in which version this issue is fixed?

**Comment by britt6612** [comment]
@muddarla-vijaya-durga right now this is on our [stable branch](https://github.com/grommet/grommet/wiki/What-is-grommet-stable-and-how-to-use-it%3F)
but planning on doing a release mid/end next week!

**Comment by muddarla-vijaya-durga** [comment]
What is the workaround for this issue?

**Comment by MikeKingdom** [comment]
> What is the workaround for this issue?

you could workaround the snapshot issue by mocking Date.now() [similar to how the above PR mocks it](https://github.com/grommet/grommet/pull/7351/files#diff-ccb0717037ab842eb18a50997804fca6fb246e12279aeab1781e923cb97b367c) so it's not random:
describe('Accordion', () => {
  beforeEach(() => {
    // Mocked values so we get consistent snapshots. I chose
    // consecutive numbers for the mocked values.
    jest.spyOn(Date, 'now').mockImplementation(() => 12345678);
    jest.spyOn(global.Math, 'random').mockReturnValue(0.123456789);
  });

  afterEach(() => {
    jest.spyOn(Date, 'now').mockRestore();
    jest.spyOn(global.Math, 'random').mockRestore();
  });

**Pull request merged**